### ISSUE TITLE
[BUG] - Clicking on subspace's proposals redirect to home page

### ISSUE THREAD

**Issue by ChaituVR** [snapshot-labs/snapshot#3541]
### Briefly describe the bug.

Proposals links of a subspace doesn't work

### How can we reproduce the bug?

1. Go to https://snapshot.org/#/2.snapspace.eth/
2. Click on any proposal of a subspace
3. It will redirect to snapshot.org

Also, in space settings page, on sub spaces section, right now we display this:
![image](https://user-images.githubusercontent.com/15967809/222252080-be622ed6-1f7c-42e5-8513-4c2273475d56.png)
But the Actual condition is: You can add some space you like to track as subspace, but subspace won't show your space if its not mutual.

**Comment by zzuziak** [comment]
For me the flow still doesn't make sense. Adding a sub-space should always be mutual, i.e. trigger a notification for the selected space (i.e. popup in the settings) to confirm that the spaces belong to the same organization. 

Otherwise why? Tracking another space shouldn't be under sub-space functionality 🤔  @ChaituVR wdyt?

**Comment by samuveth** [comment]
@zzuziak Yeah, the user experience kinda sucks and doesn't make much sense. It feels like it was just thrown together without much care. I've created an issue and added it to the Settings milestone to-do list.
